hack/update-vendor.sh

```diff
@@ -105,11 +114,13 @@ function group_replace_directives() {
      
      # otherwise print to the noreplace file
      { print > \"${go_mod_noreplace}\" }
-  "
-  cat "${go_mod_noreplace}" >  go.mod
-  echo "replace ("          >> go.mod
-  cat "${go_mod_replace}"   >> go.mod
-  echo ")"                  >> go.mod
+  " < go.mod
+  {
+    cat "${go_mod_noreplace}";
+    echo "replace (";
+    cat "${go_mod_replace}";
+    echo ")";
+  } > go.mod
```

Comment:
Here we've used a command group to redirect once, which is simpler (and preferred by shellcheck).

It has a super minor performance benefit of only needing to open the file once since bash know understands the intent to redirect all of these to the same output.

Reply:
What is the difference between `{ foo }` and `( foo )`?

Reply:
`{ foo }` is a command group but does not involve a subshell. `( foo )` is a command group in a subshell. using `{ foo; }` or:
```
{
   foo
}
```
gets the benefits of grouping without involving a subprocess / shell.

technically there are also some slightly different rules for parsing to identify them that don't apply here.